src/builder.rs

```diff
@@ -9,82 +10,110 @@ use crate::{Recorder, Worker};
+    pub fn set_distinct_id(mut self, distinct_id: Option<DistinctId>) -> Self {
+        self.distinct_id = distinct_id;
         self
     }
 
-    pub fn set_device_id(&mut self, device_id: impl Into<DeviceId>) -> &mut Self {
-        self.device_id = Some(device_id.into());
+    pub fn set_device_id(mut self, device_id: Option<DeviceId>) -> Self {
+        self.device_id = device_id;
         self
     }
 
-    pub fn set_facts(&mut self, facts: Map) -> &mut Self {
-        self.facts = Some(facts);
+    pub fn set_facts(mut self, facts: Option<Map>) -> Self {
+        self.facts = facts;
         self
     }
 
-    pub fn set_groups(&mut self, groups: Map) -> &mut Self {
-        self.groups = Some(groups);
+    pub fn set_groups(mut self, groups: Option<Map>) -> Self {
+        self.groups = groups;
         self
     }
 
     pub fn add_fact(
-        &mut self,
+        mut self,
         key: impl Into<String> + std::fmt::Debug,
         value: impl Into<serde_json::Value>,
-    ) -> &mut Self {
+    ) -> Self {
         self.facts
             .get_or_insert_with(Default::default)
             .insert(key.into(), value.into());
         self
     }
 
-    pub fn set_endpoint(&mut self, endpoint: impl Into<String>) -> &mut Self {
-        self.endpoint = Some(endpoint.into());
+    pub fn set_endpoint(mut self, endpoint: Option<String>) -> Self {
+        self.endpoint = endpoint;
         self
     }
 
-    pub fn set_timeout(&mut self, duration: impl Into<Duration>) -> &mut Self {
-        self.timeout = Some(duration.into());
+    /// Set whether reporting is enabled or disabled.
+    /// Reporting is enabled by default, but this function can be used in a pipeline for easy configuration:
+    ///
+    /// ```rust
+    /// use detsys_ids_client::builder;
+    ///
+    /// struct Cli {
+    ///   no_telemetry: bool,
+    /// }
+    ///
+    ///
+    /// # tokio_test::block_on(async {
+    ///
+    /// let cli = Cli { no_telemetry: false, };
+    ///
+    /// let (recorder, worker) = builder!()
+    ///   .set_enable_reporting(cli.no_telemetry)
+    ///   .build()
+    ///   .await
+    ///   .unwrap();
+    /// # })
+    /// ```
+    pub fn set_enable_reporting(mut self, enable_reporting: bool) -> Self {
+        self.enable_reporting = enable_reporting;
         self
     }
 
-    #[tracing::instrument(skip(self))]
-    pub async fn try_set_ssl_cert_file(
-        &mut self,
-        ssl_cert_file: impl AsRef<std::path::Path> + std::fmt::Debug,
-    ) -> Result<&mut Self, TransportsError> {
-        self.ssl_cert = Some(read_cert_file(&ssl_cert_file).await?);
-        Ok(self)
+    pub fn set_timeout(mut self, duration: Option<Duration>) -> Self {
+        self.timeout = duration;
+        self
     }
 
-    pub fn set_proxy(&mut self, proxy: Url) -> &mut Self {
-        self.proxy = Some(proxy);
+    /// Set the path to a certificate bundle.
+    ///
+    /// Note: certificate paths that are invalid or can't be parsed are ignored.
```

Comment:
If I were a consumer of this library, I'd rather know beforehand if the certificate bundle I passed was invalid, so I can react to it....

Reply:
updated the interface to let you directly pass a certificate bundle